src/services/comparator.ts

```diff
@@ -0,0 +1,389 @@
+import IntegrationDestinationService from '../interfaces/DestinationService';
+import {
+  DeliveryResponse,
+  Destination,
+  ErrorDetailer,
+  MetaTransferObject,
+  ProcessorTransformationOutput,
+  ProcessorTransformationRequest,
+  ProcessorTransformationResponse,
+  RouterTransformationRequestData,
+  RouterTransformationResponse,
+  UserDeletionRequest,
+  UserDeletionResponse,
+} from '../types';
+import tags from '../v0/util/tags';
+import stats from '../util/stats';
+import logger from '../logger';
+import { CommonUtils } from '../util/common';
+const NS_PER_SEC = 1e9;
+
+export default class ComparatorService implements IntegrationDestinationService {
+  secondaryService: IntegrationDestinationService;
+  primaryService: IntegrationDestinationService;
+
+  constructor(
+    primaryService: IntegrationDestinationService,
+    secondaryService: IntegrationDestinationService,
+  ) {
+    this.primaryService = primaryService;
+    this.secondaryService = secondaryService;
+  }
+
+  public init(): void {
+    this.primaryService.init();
+    this.secondaryService.init();
+  }
+
+  public getName(): string {
+    return 'Comparator';
+  }
+
+  public getTags(
+    destType: string,
+    destinationId: string,
+    workspaceId: string,
+    feature: string,
+  ): MetaTransferObject {
+    const metaTO = {
+      errorDetails: {
+        destType: destType.toUpperCase(),
+        module: tags.MODULES.DESTINATION,
+        implementation: this.primaryService.getName(),
+        feature,
+        destinationId,
+        workspaceId,
+        context: '[Native Integration Service] Failure During Processor Transform',
```

Comment:
Can you let me know what is the reason for having `context` as one of the labels `statTags` ?

Reply:
It is used in bugsnag, but I think this will also populate our metrics. Ok need to segregate this

Reply:
This is also fixed